src/main/kotlin/eu/theblob42/idea/whichkey/WhichKeyActionListener.kt

```diff
@@ -0,0 +1,114 @@
+package eu.theblob42.idea.whichkey
+
+import com.intellij.openapi.actionSystem.*
+import com.intellij.openapi.actionSystem.ex.AnActionListener
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.wm.WindowManager
+import com.maddyhome.idea.vim.api.injector
+import com.maddyhome.idea.vim.command.Argument
+import com.maddyhome.idea.vim.command.CommandState
+import com.maddyhome.idea.vim.command.MappingMode
+import com.maddyhome.idea.vim.impl.state.toMappingMode
+import com.maddyhome.idea.vim.newapi.vim
+import com.maddyhome.idea.vim.options.OptionAccessScope
+import eu.theblob42.idea.whichkey.config.MappingConfig
+import eu.theblob42.idea.whichkey.config.PopupConfig
+import eu.theblob42.idea.whichkey.model.Mapping
+import java.awt.event.KeyEvent
+import javax.swing.KeyStroke
+
+class WhichKeyActionListener : AnActionListener {
+    private var ignoreNextExecute = false
+
+    override fun beforeShortcutTriggered(
+        shortcut: Shortcut,
+        actions: MutableList<AnAction>,
+        dataContext: DataContext
+    ) {
```

Comment:
```suggestion
    ) {
        PopupConfig.hidePopup()
```

Make sure the popup is hidden before any processing, otherwise we might end up with lasting popup windows